config/src/config/storage_config.rs

```diff
@@ -175,6 +178,8 @@ pub struct StorageConfig {
     /// If not specificed, will use `dir` as default.
     /// Only allowed when sharding is enabled.
     pub db_path_overrides: Option<DbPathConfig>,
+    /// Waypoint used during db bootstrap to verify if the genesis txn is applied correctly
+    pub genesis_waypoint: Option<WaypointConfig>,
```

Comment:
Can we just do Option<Waypoint> instead of Option<WaypointConfig> here?

Reply:
forgot to quote and Github ate <...>, updated.

Reply:
I just don't want people to use other variants in the enum.

Reply:
I think in practise distributing a waypoint.txt is easier than embedding different strings in the config? I would keep `WaypointConfig` for now and maybe remove `FromStorage` from it separately

Reply:
The point is, in the future we probably never need to distribute storage waypoint, it's only for us to apply on our nodes.
Plus, having multiple txt files confuse people.

Reply:
I don't see much value by enforcing the limitation, while making the code less general.  If it is only us, we can hard code the waypoint in the node config. No multiple txt files in this way.